dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/compute/functions/CleanupServer.java

```diff
@@ -80,70 +80,104 @@ public Boolean apply(final String serverId) {
+               try {
+                  networkApi.deletePortList(firewallRule.destination().portList().id());
+               } catch (Throwable t) {
+                  logger.warn(format(
+                        "Failed to delete PortList(%s) associated with FirewallRule(%s) and with Server(%s). Due to - (%s)",
+                        firewallRule.destination().portList().id(), firewallRule.id(), serverId, t.getMessage()));
+               }
             }
          } else {
             logger.warn(
                   format("Server(%s) has an associated FirewallRule(%s) that was not deleted as it is in state(%s).",
                         serverId, firewallRule.id(), firewallRule.state()));
          }
+      } catch (Throwable t) {
+         logger.warn(
+               format("Failed to delete FirewallRule(%s) associated with Server(%s). Due to - (%s)", firewallRule.id(),
+                     serverId, t.getMessage()));
       }
+   }
 
-      serverApi.powerOffServer(serverId);
-      String message = format("Server(%s) not terminated within %d ms.", serverId, timeouts.nodeTerminated);
-      DimensionDataCloudControlResponseUtils
-            .waitForServerStatus(serverApi, serverId, false, true, timeouts.nodeTerminated, message);
-      serverApi.deleteServer(serverId);
-      String deleteFailureMessage = format("Server(%s) not deleted within %d ms.", serverId, timeouts.nodeTerminated);
-      waitForServerState(serverApi, serverId, State.DELETED, timeouts.nodeTerminated, deleteFailureMessage);
-      return true;
+   private void attemptDeletePublicIpBlock(final String serverId, final NetworkApi networkApi,
+         final PublicIpBlock publicIpBlock) {
+      try {
+         if (publicIpBlock.state().isNormal()) {
+            networkApi.removePublicIpBlock(publicIpBlock.id());
+         } else {
+            logger.warn(format("Server(%s) has an associated IpBlock(%s) that was not deleted as it was in state(%s).",
+                  serverId, publicIpBlock.id(), publicIpBlock.state()));
+         }
+      } catch (Throwable t) {
+         logger.warn(
+               format("Failed to delete IpBlock(%s) associated with Server(%s). Due to - (%s)", publicIpBlock.id(),
+                     serverId, t.getMessage()));
+      }
+   }
+
+   private void attemptDeleteNatRule(final String serverId, final NetworkApi networkApi, final NatRule natRule) {
+      try {
+         if (natRule.state().isNormal()) {
+            networkApi.deleteNatRule(natRule.id());
+         } else {
+            logger.warn(format("Server(%s) has an associated NatRule(%s) that was not deleted as it was in state(%s).",
+                  serverId, natRule.id(), natRule.state()));
+         }
+      } catch (Throwable t) {
+         logger.warn(
+               format("Failed to delete NatRule(%s) associated with Server(%s). Due to - (%s)", natRule.id(), serverId,
+                     t.getMessage()));
+      }
```

Comment:
Better pass the `Throwable` as the first parameter to the logger to have the complete stacktrace logged. Often just the exception message does not give enough contex.